### DISCUSSION TITLE
Does Quartz.NET 3.0 support fire & forget jobs based on events

### DISCUSSION THREAD

**Discussion by shilpa-deshpande** [quartznet/quartznet#1103] [Q&A] [answered]
Hi Team,

Currently I am using Hangfire lib with my .NET Core app (.NET version 5.0). I am looking to switch to Quartz.NET 3.0
My application has jobs triggered with cron schedules and there are some jobs which are triggered based on certain action in the application. 
I am wondering if latter part (triggering jobs once based on certain condition/action) is supported in Quartz.NET 3.0?
I am reading the documentation about SimpleTriggers and they seem to have option of SimpleTrigger.FireNow but they all look time based to me.

**Comment by lahma** [comment, marked as answer]
You can call `IScheduler.TriggerJob(JobKey)` which will create a fire once simple trigger under the hood. Make sure your job is configured as durable so that it won't be cleaned when there are no active triggers referring to it left (if that's what you want).

Out of curiosity, what's the driving force for switching library?

**Reply by shilpa-deshpande** [reply]
Thank you @lahma . 
At this point I am really exploring Quartz 3.0

**Reply by shilpa-deshpande** [reply]
@lahma  - Does Quartz.NET has any kind of monitoring. I saw there are many plugins available which can help in monitoring, but can I pass my job info to Prometheus. 
If not whats recommended plugin for monitoring?

**Reply by lahma** [reply]
There's OpenTelemetry support and example project shows that (docker containers via docker-compose). I don't have any specific recommendations, organizations usually have their chosen tools.
